### PR TITLE
specs that internal/private funcs can't be called externally

### PR DESCRIPTION
**Pull request by tomtomjhj**

### CODE REVIEW

File: gnosis/gnosis-spec.ini [hunk 1]
```diff
@@ -95,6 +95,36 @@ localMem: M
     andBool SIGS_BUF ==K selectRange(M, SIGS_LOC +Int 32, SIGS_LEN)
 +attribute: [trusted, matching(#gas)]
 
+; external call to signatureSplit should not change the state
+[signatureSplit_external]
+k: (#execute => #halt) ~> _
+output: _ => _
```
Comment (daejunpark) [inline gnosis/gnosis-spec.ini]: is it necessary?  does it return an output?

File: gnosis/gnosis-spec.ini [hunk 2]
```diff
@@ -95,6 +95,36 @@ localMem: M
...
+coinbase: _
+proxy_storage: _:IMap
++requires:
+    andBool #range(0 <= CD < 1024)
+    ; Range
+    ; bytes memory signatures
+    andBool #rangeUInt(256, SIGS_LOC)
+    andBool #rangeUInt(256, SIGS_LEN)
+    ; uint256 pos
+    andBool #rangeUInt(256, POS)
```
Comment (daejunpark) [inline gnosis/gnosis-spec.ini]: are these range conditions necessary?  what happens if we drop these?

File: gnosis/gnosis-spec.ini [hunk 3]
```diff
@@ -95,6 +95,36 @@ localMem: M
...
+[signatureSplit_external]
+k: (#execute => #halt) ~> _
+output: _ => _
+statusCode: _ => EVMC_SUCCESS
+callStack: _
+this: #PROXY_ID
+msg_sender: MSG_SENDER
+callData: #abiCallData("signatureSplit", (
+            #bytes(#buf(SIGS_LEN, SIGS)),
+            #uint256(POS) ))
```
Comment (daejunpark) [inline gnosis/gnosis-spec.ini]: can we make this more general?  like:
```
#parseByteStack(substrString(Keccak256(#generateSignature(FNAME, ARGS)), 0, 8))
```
where FNAME = `"signatureSplit"` and ARGS = `#bytes(_), #uint256(_)`?

File: gnosis/gnosis-spec.ini [hunk 4]
```diff
@@ -95,6 +95,36 @@ localMem: M
...
+localMem: .IMap => _
+pc: 0 => {STOP_EXTERNAL_CALL}
+gas: #gas(INITGAS, 0, 0) => _
+memoryUsed: 0 => _
+log: _
+refund: _
+coinbase: _
+proxy_storage: _:IMap
++requires:
+    andBool #range(0 <= CD < 1024)
```
Comment (daejunpark) [inline gnosis/gnosis-spec.ini]: the range for `CD` might be dropped?

**Comment by daejunpark** [comment]
also, please rebase on the latest master.

File: gnosis/gnosis-spec.ini (hunk 3 above)
Comment (tomtomjhj) [reply gnosis/gnosis-spec.ini]:
```
<k>
  JUMPI 408 bool2Word ( 168831172 ==K 4294967295 &Int chop ( #asWord ( 184 : 23 : 220 : 194 : #take ( 28 , WS ) ) /Int (2^224) ) ) ...
</k>
```

**Pull request merged**